### ISSUE TITLE
upgrade virtualbox

### ISSUE THREAD

**Issue by mmarseglia** [voxpupuli/puppet-virtualbox#21]
Hello,

  Is it possible to upgrade virtualbox using this module if I specify a newer version than what is currently installed?

**Comment by danzilio** [comment]
Hi @mmarseglia! Yes, you can use the `version` parameter to specify the major version you'd like to install. For example, if you'd like to install VirtualBox 5, you can set `version => '5.0'` on the `virtualbox` class.

**Comment by mmarseglia** [comment]
And that will upgrade an existing install?

**Comment by danzilio** [comment]
Sure will!

**Comment by mmarseglia** [comment]
I tried that and puppet fails.  I'm on CentOS 6.6.

Execution of '/usr/bin/yum -d 0 -e 0 -y install VirtualBox-5.0' returned 1: Transaction Check Error: file /usr/bin/VBox from install of VirtualBox-5.0-5.0.6_103037_el6-1.x86_64 conflicts with file from package VirtualBox-4.3-4.3.30_101610_el6-1.x86_64 file /usr/lib/virtualbox/VBoxAutostart from install of VirtualBox-5.0-5.0.6_103037_el6-1.x86_64 conflicts with file from package VirtualBox-4.3-4.3.30_101610_el6-1.x86_64 file

**Comment by danzilio** [comment]
Sorry about that! I just tried this and it looks like there's no direct upgrade path between major versions. In order to upgrade from 4.X to 5 you'll need to uninstall VirtualBox first. IMHO this was a rather unfortunate packaging decision on the part of the VirtualBox team.

**Comment by mmarseglia** [comment]
what do you think about adding a conditional that would uninstall version 4 if it is detected?  there is the virtualbox_version fact available.

maybe something like this pseudocode would handle upgrades and downgrades?
```
  if $manage_package then
   if detected virtualbox version matches 4.x and $version does not match 4.x then

      uninstall detected virtualbox version 
      install virtualbox normally

   else
      install virtualbox normally
  end if
end if
```

**Comment by dhollinger** [comment]
solved by #72
